packages/common/core-utils/src/heap.ts

```diff
@@ -3,6 +3,8 @@
  * Licensed under the MIT License.
  */
 
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
```

Comment:
It's better to force inline asserts, as we should have an explanation for each of why it's safe anyway and it makes it harder for new undocumented asserts to slip in.

For this file (and others like it) where the reasoning for why the non-null assert is "ok" is non-obvious, I'd personally be OK with a `// TODO: Document why this non-null assert is safe`.  Particularly looking at the bitwise operations here and their (undocumented) assumptions about calling requirements.